spec/services/threshold_waiver_service_spec.rb

```diff
@@ -1,101 +1,252 @@
 require "rails_helper"
 
 RSpec.describe ThresholdWaiverService do
-  subject(:threshold_waiver_service_response) { described_class.call(request_id, ccms_codes) }
-
   before { seed_live_data }
 
   let(:request_id) { SecureRandom.uuid }
 
-  context "when the request is successful" do
-    context "with a domestic abuse proceeding type" do
-      let(:ccms_codes) { %w[DA005] }
+  context "when the new style submission is raised" do
+    subject(:threshold_waiver_service_response) { described_class.call(request_id, values) }
+
+    context "when the request is successful" do
+      context "with a domestic abuse proceeding type" do
+        let(:values) do
+          [
+            {
+              ccms_code: "DA005",
+              client_involvement_type: "A",
+            },
+          ]
+        end
 
-      it "returns valid response with expected tasks" do
-        expect(threshold_waiver_service_response).to eq expected_da005_response
+        it "returns valid response with expected tasks" do
+          expect(threshold_waiver_service_response).to eq expected_da005_response
+        end
+      end
+
+      context "with domestic abuse and section 8 proceeding types" do
+        let(:values) do
+          [
+            {
+              ccms_code: "DA005",
+              client_involvement_type: "A",
+            },
+            {
+              ccms_code: "SE003",
+              client_involvement_type: "A",
+            },
+            {
+              ccms_code: "SE013",
+              client_involvement_type: "D",
+            },
+          ]
+        end
+
+        it "returns response with dependencies" do
+          expect(threshold_waiver_service_response).to eq expected_da005_se003_se013_response
+        end
       end
     end
 
-    context "with domestic abuse and section 8 proceeding types" do
-      let(:ccms_codes) { %w[DA005 SE003 SE013] }
+    context "when the request is unsuccessful" do
+      context "with a non_existent ccms_code" do
+        let(:values) do
+          [
+            {
+              ccms_code: "XX001",
+              client_involvement_type: "A",
+            },
+          ]
+        end
+
+        it "returns error" do
+          response = threshold_waiver_service_response
+          expect(response[:request_id]).to eq request_id
+          expect(response[:success]).to be false
+          expect(response[:error_class]).to eq "ActiveRecord::RecordNotFound"
+          expect(response[:message]).to match "Couldn't find ProceedingType"
+          expect(response[:backtrace]).to be_instance_of(Array)
```

Comment:
not a blocker: would `have_attributes` work here?
```suggestion
          expect(response).to have_attributes(request_id: request_id, success: false....
```

Reply:
I couldn't get this to work 😕